src/proto/codec.rs

```diff
@@ -131,9 +131,9 @@ impl Decoder for WebSocketProtocol {
                 ensure_buffer_has_space!(src, offset + 2);
                 // SAFETY: The ensure_buffer_has_space call has validated this
                 // A conversion from two u8s to a u16 cannot fail
-                payload_length = u16::from_be_bytes(unsafe {
-                    src.get_unchecked(2..4).try_into().unwrap_unchecked()
-                }) as usize;
+                payload_length =
+                    u16::from_be_bytes(unsafe { src.get_unchecked(2..4).try_into().unwrap() })
```

Comment:
`.try_into()` will easily prove that the slice has a length of 2, so the panic branch will be optimized out

Reply:
Correct, https://godbolt.org/z/vG1vqT3a9